### PR TITLE
Add proper error handling to export backend services

### PR DESCRIPTION
**Pull request by BuddhiWathsala**
## Purpose
$subject

## Goals
Return proper error message to the frontend. Then user can view what did happen in the console or any mechanism used in the frontend.

## Approach
Return error as a message with the response.

## Related PRs
https://github.com/siddhi-io/distribution/pull/329

## Test environment
Java version "1.8.0_201"
Java(TM) SE Runtime Environment (build 1.8.0_201-b09)

### CODE REVIEW

**Comment by BuddhiWathsala** [comment]
Note that this PR will fix this issue in backend level. https://github.com/siddhi-io/distribution/issues/377

File: tooling/components/io.siddhi.distribution.editor.core/src/main/java/io/siddhi/distribution/editor/core/internal/EditorMicroservice.java [hunk 1]
```diff
@@ -1220,14 +1228,28 @@ public Response exportApps(
                     .header("Siddhi-Docker-Key", dockerBuilderStatusKey)
                     .build();
         } catch (JsonSyntaxException e) {
-            log.error("Incorrect configuration format.", e);
+            log.error("Incorrect JSON configuration format.", e);
```
Comment (pcnfernando) [inline tooling/components/io.siddhi.distribution.editor.core/src/main/java/io/siddhi/distribution/editor/core/internal/EditorMicroservice.java]: Say the context. While exporting docker/k8

File: tooling/components/io.siddhi.distribution.editor.core/src/main/java/io/siddhi/distribution/editor/core/internal/EditorMicroservice.java [hunk 2]
```diff
@@ -1220,14 +1228,28 @@ public Response exportApps(
...
             return Response
                     .status(Response.Status.BAD_REQUEST)
+                    .entity("Incorrect JSON configuration format. " + e.getMessage())
+                    .build();
+        } catch (DockerGenerationException e) {
+            log.error("Failed docker generation. ", e);
```
Comment (pcnfernando) [inline tooling/components/io.siddhi.distribution.editor.core/src/main/java/io/siddhi/distribution/editor/core/internal/EditorMicroservice.java]: Rephrase as Exception caught while generating docker export artifacts. Update the k8 log as well

Comment (pcnfernando) [reply tooling/components/io.siddhi.distribution.editor.core/src/main/java/io/siddhi/distribution/editor/core/internal/EditorMicroservice.java]: Update response messages accordingly

File: tooling/components/io.siddhi.distribution.editor.core/src/main/java/io/siddhi/distribution/editor/core/internal/EditorMicroservice.java [hunk 3]
```diff
@@ -1287,9 +1309,10 @@ public Response getDeploymentConfigs() {
                     .type(MediaType.APPLICATION_JSON)
                     .build();
         } catch (IOException e) {
-            log.error("Cannot read deployment.yaml file", e);
+            log.error("Cannot read deployment.yaml file. ", e);
```
Comment (pcnfernando) [inline tooling/components/io.siddhi.distribution.editor.core/src/main/java/io/siddhi/distribution/editor/core/internal/EditorMicroservice.java]: Add context, what deployment.yml

**Pull request merged**